### ISSUE TITLE
When using AnimationTree, AnimationStateMachines inside BlendSpace2D don't always auto start

### ISSUE THREAD

**Issue by Jonathan-Jay-TB** [godotengine/godot#89524]
### Tested versions

- Reproducible in: 4.2 stable, 4.1 stable

### System information

Godot v4.2.1.stable - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce RTX 2060 (NVIDIA; 31.0.15.3598) - AMD Ryzen 7 4800H with Radeon Graphics (16 Threads)

### Issue description

If you use a BlendSpace2D with AnimationStateMachine blend points, specific blend points won't start on their own when in contiguous blend mode. This can be fixed by going into the state machine and pressing the start button or starting them in code, but this won't always work/persist or can be tedious depending on how many blend points there are. This issue does also carry over into builds and doesn't get fixed by pressing the start button in the state machines.

- The depth of the BlendSpace2D doesn't seem to matter, it can be the root, a node in an AnimationStateMachine, or a node in a BlendTree in an AnimationStateMachine.

- It seems to affect the blend points that aren't "in range" of the default blend value when started, so having the blend position start in one triangle would make that triangle work, and any other blend point won't. Haven't rigorously tested this though, but it seems consistent in a few different configurations.

Probably related to: #88262

Yes this can be worked around by starting each AnimationStateMachine in script, but this can be very tedious, and won't work in editor without converting your script into a tool script, which can be problematic.

### Steps to reproduce

1. Create an animation Player with a few AnimationClips that are looping moving a sprite in different directions
2. Create an AnimationTree with a BlendSpace2D
3. In the BlendSpace2D, add at least 5 AnimationStateMachines points
4. In each AnimationStateMachine, add one animation that transitions from start and press play so it's active
5. If you pressed play in each state machine, the editor preview will work
6. Write a simple script to control the blend position and then run the scene, some of the points won't work anymore
7. Reload the project, now the editor preview should have some non-functioning blend points

### Minimal reproduction project (MRP)

[AnimationTreeBug.zip](https://github.com/godotengine/godot/files/14617868/AnimationTreeBug.zip)

**Comment by TokageItLab** [comment]
Just turn on Sync.

**Comment by Jonathan-Jay-TB** [comment]
Ah didn't know that's what sync does. It's not mentioned in the docs and it didn't work when I first tried it, though I realized that I had to reload the project for it to take effect in the editor.
Could I change this into a suggestion for "promp to reload blendspace when sync is enabled" since this might not be an obvious fix for those who don't know what sync does?

> Just turn on Sync. Never related to #88262.
